processes/record_clusterer.py

```diff
@@ -0,0 +1,202 @@
+import re
+from typing import Optional
+from sqlalchemy.exc import DataError
+from logger import create_log
+from managers import DBManager, ElasticsearchManager, KMeansManager, SFRElasticRecordManager, SFRRecordManager
+from constants.get_constants import get_constants
+from model import Record, Work
+
+
+logger = create_log(__name__)
+
+
+class RecordClusterer:
+    MAX_MATCH_DISTANCE = 4
+    CLUSTER_SIZE_LIMIT = 10000
+    IDENTIFIERS_TO_MATCH = r'\|(?:isbn|issn|oclc|lccn|owi)$'
+
+    def __init__(self, db_manager: DBManager):
+        self.db_manager = db_manager
+
+        self.elastic_search_manager = ElasticsearchManager()
+
+        self.elastic_search_manager.createElasticConnection()
+        self.elastic_search_manager.createElasticSearchIngestPipeline()
+        self.elastic_search_manager.createElasticSearchIndex()
+
+        self.constants = get_constants()
+
+    def cluster_record(self, record_uuid):
+        query_filters = [
+            Record.frbr_status == 'complete',
+            Record.cluster_status == False,
+            Record.source.notin_(['oclcClassify', 'oclcCatalog']),
+            Record.title.isnot(None),
+            Record.uuid == record_uuid
+        ]
+
+        get_unclustered_record_query = self.db_manager.session.query(Record).filter(*query_filters)
+
+        unclustered_record = get_unclustered_record_query.first()
+        if not unclustered_record:
+            logger.info(f'No unclustered record found with UUID {record_uuid}')
+            return
```

Comment:
We should have the record from the record pipeline process - so method signature can be cluster_record(self, record: Record)

Then we don't need this logic!